frontend/src/components/ConstraintForm/SubjectAccordionItem.vue

```diff
@@ -59,14 +74,16 @@ export default {
       <span>Teacher</span>
       <select v-model="subject.teacher.name" class="rsprd-select">
         <option value="">-</option>
-        <option v-for="teacher in teachers" :key="teacher">{{ teacher }}</option>
+        <option v-for="teacher in teachers" :key="teacher">
+          {{ getTeacherOptionContent(teacher) }}
```

Comment:
It is usually best to do this type of processing outside of the template. Maybe after fetching the data. The reason is how methods are handled in templates. Vue can not cache results of methods in called in the template so it calls them again on each re-render. Se more [here](https://vuejs.org/guide/essentials/computed.html#computed-caching-vs-methods).